packages/mwp-core/src/renderers/server-render.jsx

```diff
@@ -277,26 +277,25 @@ const makeRenderer = (
 		};
 		const checkReady = state =>
 			state.preRenderChecklist.every(isReady => isReady);
-		const populateStore = store =>
-			new Promise((resolve, reject) => {
+		const populateStore = store => {
+			return addFlags(store).then(() => {
 				// dispatch SERVER_RENDER to kick off API middleware
 				store.dispatch({ type: SERVER_RENDER, payload: url });
 
-				if (checkReady(store.getState())) {
-					addFlags(store).then(() => {
-						resolve(store);
+				return new Promise((resolve, reject) => {
+					// dispatch SERVER_RENDER to kick off API middleware
+					store.dispatch({ type: SERVER_RENDER, payload: url });
+					const unsubscribe = store.subscribe(() => {
```

Comment:
before setting up the `store.subscribe()`, it's important to check whether the store is already ready (synchronously) so that you can `resolve` immediately, similar to what was happening before on line 285.

If it's not ready synchronously, then you set up the `store.subscribe()` listener to `resolve` async.